src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -3382,4 +3519,87 @@ private String formatTestCasePhase(String runPhase, int testCase)
     {
         return(String.format("%10s test case %s", runPhase, StringUtilities.rpad(' ', 5, Integer.toString(testCase) + "...")));
     }
+    
+    /**
+     * Get the filename of a judge's file.  This could be a full path if the data files are external.  If the
+     * files are internal, then we return the first one's name since each dataset will get copied to the name
+     * of the first item.
+     * 
+     * @return filename of a judge's file
+     * @param wantInput - boolean indicating if we want the input file (true).  If false, we want the answer file
+     * @param setIndex - Which dataset element number are we interested in [0, #datasets-1]
+     */
+    private String getJudgeFileName(Utilities.DataFileType type, int setIndex)
+    {
+        String result = null;
+        SerializedFile serializedFile = null;
+        
+        try {
+            // it's a little more work for external files
+            if (problem.isUsingExternalDataFiles()) {
+                
+                if(type == Utilities.DataFileType.JUDGE_DATA_FILE) {
+                    serializedFile = problemDataFiles.getJudgesDataFiles()[setIndex];
+                } else {
+                    serializedFile = problemDataFiles.getJudgesAnswerFiles()[setIndex];               
+                }
+                //Note: last argument (type) is unused in locateJudgesDataFile
+                result = Utilities.locateJudgesDataFile(problem, serializedFile, getContestInformation().getJudgeCDPBasePath(), type);
+            } else {
+                // For internal files, the appropriate data files are copied to the FIRST datafile's name in the
+                // execute folder, so we always return that one.
+                if(type == Utilities.DataFileType.JUDGE_DATA_FILE) {
+                    result = prefixExecuteDirname(problem.getDataFileName());
+                } else {
+                    result = prefixExecuteDirname(problem.getAnswerFileName());
+                }
+            }
+        } catch (Exception e)
+        {
+            //if we got far enough to get the serialized file, show the expected name in the log message
+            if(serializedFile != null) {
+                log.log(Log.WARNING, "Can not get " + type.toString() + " expected filename (" + serializedFile.getName() + ") for dataset " + (setIndex+1) + ": " + e.getMessage(), e);
+            } else {
+                log.log(Log.WARNING, "Can not get " + type.toString() + " filename for dataset " + (setIndex+1) + ": " + e.getMessage(), e);
+            }
+        }
+        return(result);
+    }
+    
+    /**
+     * Unpacks and creates a custom validator program.  If any error happens, an exception is thrown
+     * 
+     * @throws SecurityException - if the copy of the output validator can not be generated in the execute folder
+     * @throws IllegalStateException - if not output validator is available
+     */
+    void createValidatorProgram()
```

Comment:
This method declaration should have `private` on it.

Reply:
Will be fixed in next commit.